pupil-api/src/services/pupil-authentication.service.ts

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+import * as azureStorage from 'azure-storage'
+import * as bluebird from 'bluebird'
+import { clone } from 'ramda'
+import * as winston from 'winston'
+
+let azureTableService: any
+const authTable = 'preparedCheck'
+
+export const pupilAuthenticationService = {
+  authenticate: async function authenticate (pupilPin: string, schoolPin: string, tableService?: any): Promise<object> {
+    // set tableService to azureTableService if not provided, but only instantiate it once.
+    if (!tableService) {
+      if (!azureTableService) {
+        azureTableService = azureStorage.createTableService()
+        bluebird.promisifyAll(azureTableService, {
+          promisifier: (originalFunction) => function (...args) {
+            return new Promise((resolve, reject) => {
+              try {
+                originalFunction.call(this, ...args, (error, result, response) => {
+                  if (error) {
+                    return reject(error)
+                  }
+                  resolve({ result, response })
+                })
+              } catch (error) {
+                reject(error)
+              }
+            })
+          }
+        })
+      }
+      tableService = azureTableService
+    }
+
+    const { result } = await tableService.retrieveEntityAsync(authTable, schoolPin, pupilPin)
+    if (result.isCollected._ === true) {
+      throw new Error('Repeat attempts to collect data are not allowed')
+    }
+
+    // Prepare the pupil data for use by the SPA
+    const data = this.preparePupilData(result)
+
+    // Mark the data as having been collected
+    try {
+      await this.markAsCollected(tableService, clone(result))
```

Comment:
i imagine the difference in performance could be significant if we don't make this call.

We could move away from this by invalidating/deleting the record when processing the `check-started` queue message

Reply:
Possibly there would be a slight difference.  I didn't see any difference when running a single sample of each, on the contrary, the one without the update took exactly the same time.  Looks like it would be worth running a performance test to see if the update call if a hotspot.